tests/ssg_test_suite/common.py

```diff
@@ -289,21 +499,65 @@ def iterate_over_rules():
             id -- full rule id as it is present in datastream
             short_id -- short rule ID, the same as basename of the directory
                         containing the test scenarios in Bash
-            files -- list of executable .sh files in the "tests" directory
+            files -- list of executable .sh files in the uploaded tarball
     """
-    for dirpath, dirnames, filenames in walk_through_benchmark_dirs():
+
+    # Here we need to perform some magic to handle parsing the rule (from a
+    # product perspective) and loading any templated tests. In particular,
+    # identifying which tests to potentially run involves invoking the
+    # templating engine.
+    #
+    # Begin by loading context about our execution environment, if any.
+    product_yaml = get_product_context(product)
+
+    # Initialize a mock template_builder.
+    empty = "/ssgts/empty/placeholder"
+    template_builder = ssg.templates.Builder(product_yaml, empty,
+        _SHARED_TEMPLATES, empty, empty)
+
+    for dirpath, dirnames, filenames in walk_through_benchmark_dirs(product):
         if "rule.yml" in filenames and "tests" in dirnames:
             short_rule_id = os.path.basename(dirpath)
+
+            # Load the rule itself to check for a template.
+            rule, local_env_yaml = load_rule_and_env(dirpath, product_yaml, product)
+
+            # All tests is a mapping from path (in the tarball) to contents
+            # of the test case. This is necessary because later code (which
+            # attempts to parse headers from the test case) don't have easy
+            # access to templated content. By reading it and returning it
+            # here, we can save later code from having to understand the
+            # templating system.
+            all_tests = dict()
+
+            # Start
+            if rule.template:
+                templated_tests = template_builder.get_all_tests(
+                    rule.id_, rule.template, local_env_yaml)
+                all_tests.update(templated_tests)
+
+            # Add additional tests from the local rule directory. Note that,
+            # like the behavior in template_tests, this will overwrite any
+            # templated tests with the same file name.
             tests_dir = os.path.join(dirpath, "tests")
             tests_dir_files = os.listdir(tests_dir)
+            for test_case in tests_dir_files:
+                test_path = os.path.join(tests_dir, test_case)
+                if os.path.isdir(test_path):
+                    continue
+
+                all_tests[test_case] = process_file(test_path, local_env_yaml)
```

Comment:
Forgot this the first time. :-) But I was able to use this to update some package install clauses in some rules, e.g.:

```diff
diff --git a/linux_os/guide/system/auditing/service_auditd_enabled/tests/disabled.fail.sh b/linux_os/guide/system/auditing/service_auditd_enabled/tests/disabled.fail.sh
index 158ffcfc39..a2f6f6c60e 100644
--- a/linux_os/guide/system/auditing/service_auditd_enabled/tests/disabled.fail.sh
+++ b/linux_os/guide/system/auditing/service_auditd_enabled/tests/disabled.fail.sh
@@ -1,5 +1,10 @@
 #!/bin/bash
+
+{{% if 'audit' in platform_package_overrides %}}
+# packages = {{{ platform_package_overrides['audit'] }}}
+{{% else %}}
 # packages = audit
+{{% endif %}}
 
 systemctl stop auditd
 systemctl disable auditd
```

And it works on Ubuntu. :-)

Reply:
(in #7295 we provide a better macro for this).